src/NaCl.Core/Base/Salsa20Base.cs

```diff
@@ -29,58 +44,21 @@ public abstract class Salsa20Base : Snuffle
         /// <param name="state">The state.</param>
         /// <param name="nonce">The nonce.</param>
         /// <param name="counter">The counter.</param>
-        protected abstract void SetInitialState(Span<uint> state, ReadOnlySpan<byte> nonce, int counter);
+        internal protected abstract void SetInitialState(Span<uint> state, ReadOnlySpan<byte> nonce, int counter);

```

Comment:
Something minor... the access modifier should be `protected internal`, just need to fix the ordering.